File: src/main/java/com/dingdangmaoup/resin/pura/resin/common/MBeanUtil.java
package com.dingdangmaoup.resin.pura.resin.common;

import javax.management.MalformedObjectNameException;
import javax.management.ObjectName;
import org.jetbrains.annotations.NonNls;

public final class MBeanUtil {

  private MBeanUtil() {

  }

  public static ObjectName newObjectName(@NonNls String objectName) {
    try {
      return ObjectName.getInstance(objectName);
    }
    catch (MalformedObjectNameException e) {
      throw new InternalError("Never happens");
    }
  }
}

File: src/main/kotlin/com/dingdangmaoup/resin/pura/resin/common/MBeanUtil.kt
package com.dingdangmaoup.resin.pura.resin.common

import javax.management.MalformedObjectNameException
import javax.management.ObjectName

object MBeanUtil {
    @JvmStatic
    fun newObjectName(objectName: String): ObjectName {
        return try {
            ObjectName.getInstance(objectName)
        } catch (_: MalformedObjectNameException) {
            throw InternalError("Never happens")
        }
    }
}
